### PR TITLE
#175494074 ft:Users should be able to make multi city request

### PR DESCRIPTION
**Pull request by tigthor**
[start 175494074]

#### What does this PR do?
- the requester should be able to make one request that has multiple destinations embedded.
- This request should then be sent to the requester's manager for approval.
- create multi city trip
#### Description of Task to be completed?
I have this endpoint working
`POST /api/trip/multi-city`
#### How should this be manually tested?
- open your terminal and clone this repo by running `git clone https://github.com/Soma-Technologies-Inc/visitAfrica-backend` 
- run `git checkout ft-multicityrequest-175494074` to navigate to my branch
- run `git pull` to make sure you have updated code
- run `npm install` to install all needed packages
- run `npm run migrate` to create migrations for the database
- navigate to login route `POST /api/auth/login` and login to get token that will b eused as a header for multi city request
- use these body data `{
    "email": "loua@gmail.com",
    "password": "Loua@123"
}`
- you will get a jwt then
- navigate to multi city request route `POST /api/trip/multi-city` and use a body data like this `{
   "departureFrom": "Romania",
   "departureTo": "Budapest",
   "startingDate": "2020-12-20",
   "returningDate": "2020-12-30",
   "multiCity": [
      {
         "name": "Budapest",
         "startingDate": "2020-12-20",
         "returningDate": "2020-12-30"
      },
      {
         "name": "Budapest",
         "startingDate": "2020-12-20",
         "returningDate": "2020-12-30"
      },
      {
         "name": "Budapest",
         "startingDate": "2020-12-20",
         "returningDate": "2020-12-30"
      }
   ],
   "reason": "travel"
}`
- before making a request append a `authorisation` header which have value `bearer token` replace token with the jwt token you got from `login` route
- then if you did everything correctly you will see output similar to screeenshots which are below 
#### Screenshots

Screenshot at 2020-12-19 00-07-15
Screenshot at 2020-12-19 00-07-45
Screenshot at 2020-12-19 00-08-08

#### Relevant Pivot tracker Id
[#175494074](https://www.pivotaltracker.com/story/show/175494074)

### CODE REVIEW

**Comment by Niyonsengaeric** [comment]
@tigthor, please make a correction on What does this PR do? session.

- is this the right status code?

403 setting authorization

- change your request body to be a JSON object, not an array.
-  @tigthor  can you make it one request with one departure date and return date with those multi-stop on it.

**Comment by tigthor** [comment]
> @tigthor, please make a correction on What does this PR do? session.
> 
> * is this the right status code?
> 
> 403 setting authorization
> 
> * change your request body to be a JSON object, not an array.
> * @tigthor  can you make it one request with one departure date and return date with those multi-stop on it.

Yes I am gonna work on that

**Comment by Liliane003** [comment]
LGTM

File: src/database/migrations/20201129004549-create-user.js [hunk 1]
```diff
@@ -47,6 +47,10 @@ export async function up(queryInterface, Sequelize) {
 			type: Sequelize.STRING,
 			defaultValue: 'avatar.jpg',
 		},
+		lineManagerId: {
```
Comment (Niyonsengaeric) [inline src/database/migrations/20201129004549-create-user.js]: why are you hardcoding lineManager?

**Review by Niyonsengaeric — CHANGES_REQUESTED** [review]

**Pull request merged**